src/gha_logs.rs

```diff
@@ -234,6 +227,93 @@ async fn process_logs(
         }
     };
 
+    let tree_roots = tree_roots.iter().map(|p| regex::escape(p)).join("|");
+    let regex = format!(
+        r#"(?<boundary>[^a-zA-Z0-9.\\/])(?<inner>(?:[\\/]?(?:checkout[\\/])?(?<path>(?:{tree_roots})(?:[\\/][a-zA-Z0-9_$\\-.\\/]+)?))(?::(?<line>[0-9]+):(?<col>[0-9]+))?)"#,
+    );
+    let gha_logs_path_regex = Regex::new(&regex).unwrap();
+
+    // 1. Remove UTF-8 useless BOM
+    let logs = logs.strip_prefix('\u{FEFF}').unwrap_or(logs.as_str());
+
+    // 2. Tranform the ANSI escape codes to HTML
+    let anstyle_svg::HtmlParts { body, style } = anstyle_svg::Term::new()
+        .use_html5(true)
+        .render_html_parts(logs);
+
+    // 3. Add a self-referencial anchor to all timestamps at the start of the lines
+    let html = TIMESTAMP_REGEX.replace_all(&body, |ts: &regex::Captures| {
+        let ts = &ts[0][1..];
+        format!("><a id='{ts}' href='#{ts}' class='timestamp'>{ts}</a>")
```

Comment:
Is the leading `>` intended?

Reply:
Yes, I use the closing of the `<br>` tag as "anchor" to ensure it's a timestamp we want to replace and not something part of the log.

Reply:
I see, right, there's no lookbehind in `regex` yet. Please add a comment on top of `TIMESTAMP_REGEX` to explain what is it looking for.

Reply:
Done!